pkg/controller/storagecluster/storagecluster.go

```diff
@@ -1177,20 +1178,14 @@ func (c *Controller) getCurrentMaxStorageNodesPerZone(
 
 // getDefaultMaxStorageNodesPerZone aims to return a good value for MaxStorageNodesPerZone with the
 // intention of having at least 3 nodes in the cluster.
-func getDefaultMaxStorageNodesPerZone(zoneMap map[string]uint64) uint32 {
-	numZones := len(zoneMap)
-	switch numZones {
-	case 0, 1:
-		// If there is a single Zone, have all 3 nodes in the same zone
-		return 3
-	case 2:
-		// If there are two zones, it'll be tricky since we'll always lose quorum when a zone
-		// goes down. Let's have 2 nodes in a zone so that we have a 4 node cluster.
-		return 2
-	default:
-		// In a cluster with 3 or more zones, let's have one node in each zone.
-		return 1
+func getDefaultMaxStorageNodesPerZone(nodeList *v1.NodeList) (uint32, error) {
+	zoneMap := getZoneMap(nodeList)
+	numZones := uint64(len(zoneMap))
+	storageNodes, err := storagenodescount.GetStorageNodesCount(uint64(len(nodeList.Items)), numZones)
```

Comment:
this is the new function you added in this call right? else nothing got changed when setting the default value?

Reply:
This is an old function with new parameters. Yes, other than the content of this function nothing much has changed. Just that the caller should be able to pass in the nodeList